src/components/order/__tests__/ReceiverForm.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { describe, it, expect, vi } from "vitest";
+import ReceiverForm from "@/components/order/ReceiverForm";
+import type { Receiver } from "@/types/order";
+
+describe("ReceiverForm", () => {
+  const ReceiverList: Receiver[] = [
+    { name: "가나다", phone: "010-1234-5678", quantity: 1 },
+    { name: "마바사", phone: "010-9876-5432", quantity: 2 },
+  ];
+
+  const defaultProps = {
+    receiverList: [] as Receiver[],
+    setReceiverList: vi.fn(),
+  };
+
+  it("받는 사람 목록이 비어있을 때 비어있다는 메세지 표시", () => {
+    render(<ReceiverForm {...defaultProps} />);
+
+    expect(screen.getByText("받는 사람이 없습니다.")).toBeInTheDocument();
+    expect(screen.getByText("받는 사람을 추가해주세요.")).toBeInTheDocument();
+  });
+
+  it("받는 사람 목록이 있을 때 받는 사람 리스트 표로 표시", () => {
+    render(<ReceiverForm {...defaultProps} receiverList={ReceiverList} />);
+
+    expect(screen.getByText("이름")).toBeInTheDocument();
+    expect(screen.getByText("전화번호")).toBeInTheDocument();
+    expect(screen.getByText("수량")).toBeInTheDocument();
+
+    expect(screen.getByText("가나다")).toBeInTheDocument();
+    expect(screen.getByText("010-1234-5678")).toBeInTheDocument();
+    expect(screen.getByText("1")).toBeInTheDocument();
+
+    expect(screen.getByText("마바사")).toBeInTheDocument();
+    expect(screen.getByText("010-9876-5432")).toBeInTheDocument();
+    expect(screen.getByText("2")).toBeInTheDocument();
+  });
+
+  it("받는 사람 목록이 있을 때 버튼이 '수정'으로 변경", () => {
+    render(<ReceiverForm {...defaultProps} receiverList={ReceiverList} />);
+
+    expect(screen.getByText("수정")).toBeInTheDocument();
+    expect(screen.queryByText("추가")).not.toBeInTheDocument();
+  });
+
+  it("추가 버튼을 클릭하면 모달 열림", () => {
+    render(<ReceiverForm {...defaultProps} />);
+
+    const addButton = screen.getByText("추가");
+    fireEvent.click(addButton);
+
+    expect(screen.getByTestId("receiver-modal")).toBeInTheDocument();
```

Comment:
개인적으로 테스트코드를 작성할 때, given/when/then 패턴을 활용하는데요

이때, 주석으로 given when then을 남겨주면 테스트코드를 읽기가 훨씬 편해지더라구요.

한번 고려해보시면 좋을 것 같습니다.
```suggestion
    // given
    render(<ReceiverForm {...defaultProps} />);

    // when - 추가 버튼 클릭
    const addButton = screen.getByText("추가");
    fireEvent.click(addButton);
   
    // then - 모달 오픈 확인
    expect(screen.getByTestId("receiver-modal")).toBeInTheDocument();
```